Kitodo/src/main/java/org/kitodo/production/services/command/ImportProcesses.java

```diff
@@ -0,0 +1,345 @@
+     * {@code indir} directory must be specified, it must exist, and the Tomcat
+     * must have permission to get the directory listing (execution permission).
+     * 
+     * @param indir
+     *            directory entered by the user. Can be {@code null} if the
+     *            parameter is not specified
+     * @return a Path object to the root directory for bulk import
+     */
+    private final Path checkIndir(String indir) {
+        if (Objects.isNull(indir)) {
+            throw new IllegalArgumentException("kitodoScript.importProcesses.indir.isNull");
+        }
+        Path importRoot = Paths.get(indir);
+        if (!Files.isDirectory(importRoot)) {
+            throw new IllegalArgumentException("kitodoScript.importProcesses.indir.isNoDirectory");
+        }
+        if (!Files.isExecutable(importRoot)) {
+            throw new IllegalArgumentException("kitodoScript.importProcesses.indir.cannotExecute");
+        }
+        return importRoot;
+    }
+
+    /**
+     * Checks whether the parameter {@code project} is specified and valid. If
+     * not, a corresponding error message is thrown as an exception. (This is
+     * picked up above and will be translated and displayed to the user.) The
+     * project must be specified, it must be syntactically valid (a positive
+     * integer), and a project with that ID must exist.
+     * 
+     * @param project
+     *            user-entered project number. Can be {@code null} if the
+     *            parameter is not specified
+     * @return the project object from the database
+     */
+    private final Project checkProject(String project) {
+        if (Objects.isNull(project)) {
+            throw new IllegalArgumentException("kitodoScript.importProcesses.project.isNull");
+        }
+        if (!project.matches("[\\d]+")) {
+            throw new IllegalArgumentException("kitodoScript.importProcesses.project.isNoProjectID");
+        }
+        Integer projectInteger = Integer.valueOf(project);
+        try {
+            return projectService.getById(projectInteger);
+        } catch (DAOException e) {
+            logger.catching(e);
+            throw new IllegalArgumentException("kitodoScript.importProcesses.project.noProjectWithID");
+        }
+    }
+
+    /**
+     * Checks whether the parameter {@code template} is specified and valid. If
+     * not, a corresponding error message is thrown as an exception. (This is
+     * picked up above and will be translated and displayed to the user.) The
+     * production template must be specified, its ID must be syntactically valid
+     * (a positive integer), and a production template with that ID must exist.
+     * 
+     * @param project
+     *            user-entered production template number. Can be {@code null}
+     *            if the parameter is not specified
+     * @return the production template object from the database
+     */
+    private final Template checkTemplate(String template) {
```

Comment:
```suggestion
    private Template checkTemplate(String template) {
```
My IDE tells me: 
`Since private methods cannot be meaningfully overridden because of their visibility, declaring them final is redundant.`